src/main/scala/pureconfig/WordDelimiter.scala

```diff
@@ -0,0 +1,42 @@
+package pureconfig
+
+trait WordDelimiter {
+  def toTokens(s: String): Seq[String]
```

Comment:
Instead of restricting config<->field mappings to word-based rules, wouldn't it be more flexible if you required only an object able to convert from field names to config keys and vice-versa? `WordDelimiter` could then be a subclass that implemented the conversion via word tokenization.

The use case I have in mind is when users have config schemas they do not control and they want to do fully custom mappings of specific keys. Of course common sense suggests that such flexibility should not be abused, but it would be good to have that choice if the need arises.